File: part3/functionalProgramming/recursion.cpp
#include <stdio.h>
long int fact(int n);

int main() {
    int n;
    printf("输入正整数: ");
    scanf("%d", &n);
    printf("Factorial of %d = %ld", n, fact(n));
    return 0;
}
long int fact(int n) {
    if (n >= 1)
        return n*fact(n-1);
    else
        return 1;
}

File: part3/functionalProgramming/recursion.py
def fact(n):
    if n == 1:
        return n
    else:
        return n* fact (n-1)

    # accepts input from user
num = int(input("Enter a number: "))
# check whether number is positive or not

if num < 0:
    print("Sorry, factorial does not exist for negative numbers")
else:
    print("The factorial of " + str(num) +  " is " + str(fact(num)))
